pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1401,34 +1295,28 @@ func podToMemberPod(pod *v1.Pod, includeIP, includePodRef bool) *controlplane.Gr
 	return memberPod
 }
 
-func externalEntityToGroupMember(ee *v1alpha1.ExternalEntity) *controlplane.GroupMember {
+func externalEntityToGroupMember(ee *v1alpha2.ExternalEntity) *controlplane.GroupMember {
 	memberEntity := &controlplane.GroupMember{}
-	for _, endpoint := range ee.Spec.Endpoints {
-		var namedPorts []controlplane.NamedPort
-		for _, port := range endpoint.Ports {
-			namedPorts = append(namedPorts, controlplane.NamedPort{
-				Port:     port.Port,
-				Name:     port.Name,
-				Protocol: controlplane.Protocol(port.Protocol),
-			})
-		}
-		ep := controlplane.Endpoint{
-			IP:    ipStrToIPAddress(endpoint.IP),
-			Ports: namedPorts,
+	namedPorts := make([]controlplane.NamedPort, len(ee.Spec.Ports))
+	var ipAddr []controlplane.IPAddress
+	for i, port := range ee.Spec.Ports {
+		namedPorts[i] = controlplane.NamedPort{
+			Port:     port.Port,
+			Name:     port.Name,
+			Protocol: controlplane.Protocol(port.Protocol),
 		}
-		memberEntity.Endpoints = append(memberEntity.Endpoints, ep)
 	}
-	entityRef := controlplane.ExternalEntityReference{
-		Name:      ee.Name,
-		Namespace: ee.Namespace,
+	for _, ep := range ee.Spec.Endpoints {
+		ipAddr = append(ipAddr, ipStrToIPAddress(ep.IP))
 	}
-	memberEntity.ExternalEntity = &entityRef
+	memberEntity.Ports = namedPorts
+	memberEntity.IPs = ipAddr
```

Comment:
I think `memberEntity.ExternalEntity` should be set?

Reply:
Thanks for spotting this. Done.